extensions/toggl-track/src/index.tsx

```diff
@@ -115,6 +130,44 @@ function ListView() {
                       </ExtensionContextProvider>
                     }
                   />
+                  <ActionPanel.Section>
+                    <Action
+                      title="Delete Time Entry"
+                      icon={Icon.Trash}
+                      style={Action.Style.Destructive}
+                      shortcut={{ modifiers: ["ctrl"], key: "x" }}
+                      onAction={async () => {
+                        await confirmAlert({
+                          title: "Delete Time Entry",
+                          message: "Are you sure you want to delete this time entry?",
+                          icon: {
+                            source: Icon.Trash,
+                            tintColor: Color.Red,
+                          },
+                          primaryAction: {
+                            title: "Delete",
+                            style: Alert.ActionStyle.Destructive,
+                            onAction: async () => {
+                              const toast = await showToast({
+                                style: Toast.Style.Animated,
+                                title: "Deleting time entry",
+                              });
+                              try {
+                                await mutateTimeEntries(removeTimeEntry(timeEntry.workspace_id, timeEntry.id));
+
+                                toast.style = Toast.Style.Success;
+                                toast.title = "Successfully deleted time entry 🎉";
+                              } catch (error) {
+                                toast.style = Toast.Style.Failure;
+                                toast.title = "Failed to delete time entry 😥";
+                                toast.message = error instanceof Error ? error.message : undefined;
+                              }
```

Comment:
I'd recommend using `withToast` here. It will clean up the code and make things more consistent. You can find examples of how in [`src/components/ClientForm.tsx`](https://github.com/raycast/extensions/blob/91553c40e190f0094b6d6a75b8dca4b4ebf5c627/extensions/toggl-track/src/components/ClientForm.tsx#L28).

Reply:
This can be resolved.